addons/mail/models/mail_scheduled_message.py

```diff
@@ -172,27 +172,43 @@ def _post_message(self, raise_exception=True):
             still post permission on the related record, and to allow for the attachments to be
             transferred to the messages (see _process_attachments_for_post in mail.thread)
             if raise_exception is set to False, the method will skip the posting of a message
-            instead of raising an error. This is useful when scheduled messages are sent from
-            the _post_messages_cron
+            instead of raising an error, and send a notification to the author about the failure.
+            This is useful when scheduled messages are sent from the _post_messages_cron.
         """
         notification_parameters_whitelist = self._notification_parameters_whitelist()
         for scheduled_message in self:
             message_creator = scheduled_message.create_uid
             try:
-                scheduled_message.with_user(message_creator)._check()
-            except AccessError:
+                with self.env.cr.savepoint():
+                    scheduled_message.with_user(message_creator)._check()
+                    self.env[scheduled_message.model].browse(scheduled_message.res_id).with_user(message_creator).message_post(
+                        attachment_ids=list(scheduled_message.attachment_ids.ids),
+                        author_id=scheduled_message.author_id.id,
+                        body=scheduled_message.body,
+                        partner_ids=list(scheduled_message.partner_ids.ids),
+                        subtype_xmlid='mail.mt_note' if scheduled_message.is_note else 'mail.mt_comment',
+                        **{k: v for k, v in json.loads(scheduled_message.notification_parameters or '{}').items() if k in notification_parameters_whitelist},
+                    )
+            except Exception as e:
                 if raise_exception:
                     raise
-                _logger.info("Posting of scheduled message %s failed: user %s cannot post on the record", scheduled_message.id, message_creator.id)
-                continue
-            self.env[scheduled_message.model].browse(scheduled_message.res_id).with_user(message_creator).message_post(
-                attachment_ids=list(scheduled_message.attachment_ids.ids),
-                author_id=scheduled_message.author_id.id,
-                body=scheduled_message.body,
-                partner_ids=list(scheduled_message.partner_ids.ids),
-                subtype_xmlid='mail.mt_note' if scheduled_message.is_note else 'mail.mt_comment',
-                **{k: v for k, v in json.loads(scheduled_message.notification_parameters or '{}').items() if k in notification_parameters_whitelist},
-            )
+                _logger.info("Posting of scheduled message with ID %s failed for the following reason: %s", scheduled_message.id, e)
```

Comment:
using `%s` to log exceptions makes for pretty bad messages:
```
$ python3
Python 3.10.12 (main, Nov  6 2024, 20:22:13) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
~/.python_startup loaded => report()
>>> import logging
>>> logging.basicConfig()
>>> try:
...     {}["jexistepas"]
... except Exception as exc:
...     logging.warning("Error: %s", exc)
... 
WARNING:root:Error: 'jexistepas'
```

If you want to keep it very short, you can log a single frame:
```suggestion
                _logger.info("Posting of scheduled message with ID %s failed\n%s", scheduled_message.id, traceback.format_exc(limit=-1))
```
-1 make it log only the last frame (the one closest to the error)

otherwise if logging everything is ok
```suggestion
                _logger.info("Posting of scheduled message with ID %s failed", scheduled_message.id, exc_info=True)
```

Reply:
Hey! thanks for the validation 🙂
I tried your suggestions but it makes runbot unhappy (because it finds a traceback in the logs when running the test that I added that checks the behaviour when there was an error during the posting of a scheduled message).
Do you know any way to prevent this? or are we doomed to keep the bad log message 😅

Reply:
You can use a mute logger in tests so that it is not displayed in the test, and runbot is therefore happy.